src/action-ros-ci.ts

```diff
@@ -125,32 +136,51 @@ async function run() {
 		const packageNameList = packageName.split(RegExp("\\s"));
 		const rosWorkspaceName = "ros_ws";
 		const rosWorkspaceDir = path.join(workspace, rosWorkspaceName);
-		const sourceRosBinaryInstallation = core.getInput(
-			"source-ros-binary-installation"
-		);
-		const sourceRosBinaryInstallationList = sourceRosBinaryInstallation
-			? sourceRosBinaryInstallation.split(RegExp("\\s"))
-			: [];
-
+		const targetRos1Distro = core.getInput("target-ros1-distro");
+		const targetRos2Distro = core.getInput("target-ros2-distro");
 		const vcsRepoFileUrlListAsString = core.getInput("vcs-repo-file-url") || "";
 		const vcsRepoFileUrlList = vcsRepoFileUrlListAsString.split(RegExp("\\s"));
-		const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter(x => x != "");
-		const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map(x =>
+		const vcsRepoFileUrlListNonEmpty = vcsRepoFileUrlList.filter(
+			(x) => x != ""
+		);
+		const vcsRepoFileUrlListResolved = vcsRepoFileUrlListNonEmpty.map((x) =>
 			resolveVcsRepoFileUrl(x)
 		);
 
 		const coverageIgnorePattern = core.getInput("coverage-ignore-pattern");
 
 		let commandPrefix = "";
-		if (sourceRosBinaryInstallation) {
+		if (targetRos1Distro) {
 			if (process.platform !== "linux") {
 				core.setFailed(
 					"sourcing binary installation is only available on Linux"
 				);
 				return;
 			}
-			for (let rosDistribution of sourceRosBinaryInstallationList) {
-				commandPrefix += `source /opt/ros/${rosDistribution}/setup.sh && `;
+			if (validDistro.indexOf(targetRos1Distro) <= -1) {
+				return false;
+			}
+			commandPrefix += `source /opt/ros/${targetRos1Distro}/setup.sh && `;
+		}
+		if (targetRos2Distro) {
+			if (process.platform !== "linux") {
+				core.setFailed(
+					"sourcing binary installation is only available on Linux"
+				);
+				return;
+			}
+			if (validDistro.indexOf(targetRos2Distro) <= -1) {
+				return false;
+			}
+			commandPrefix += `source /opt/ros/${targetRos2Distro}/setup.sh && `;
+		}
+
+		if (process.platform == "linux") {
```

Comment:
These arguments are not explicitly around "binary installations" - they are just the target distro of the build, so we should change this language to "No ROS1 or ROS2 distribution target specified for the build."

Additionally, it may be a bit more readable if this check is up higher before the "source" logic.